### PR TITLE
DEVENGAGE-1719: Update the GO SDK so that I could use the new role lo…

### PR DESCRIPTION
**Pull request by carnellj-genesys**
…ok up endpoint with the usecount disabled.

### CODE REVIEW

File: genesyscloud/data_source_genesyscloud_routing_email_domain.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func dataSourceRoutingEmailDomainRead(ctx context.Context, d *schema.ResourceDat
 		for pageNum := 1; ; pageNum++ {
 			const pageSize = 100
 
-			domains, _, getErr := routingAPI.GetRoutingEmailDomains(pageSize, pageNum, false)
+			domains, _, getErr := routingAPI.GetRoutingEmailDomains(pageSize, pageNum, false, "")
```
Comment (carnellj-genesys) [inline genesyscloud/data_source_genesyscloud_routing_email_domain.go:40]: Email API added a new query parameter so had to adjust the API call.

File: genesyscloud/resource_genesyscloud_routing_queue.go [hunk 2]
```diff
@@ -595,22 +595,25 @@ func readQueue(ctx context.Context, d *schema.ResourceData, meta interface{}) di
...
-				d.Set("media_settings_chat", flattenMediaSetting(chatSettings))
-			}
-			if emailSettings, ok := (*currentQueue.MediaSettings)[mediaSettingsKeyEmail]; ok {
-				d.Set("media_settings_email", flattenMediaSetting(emailSettings))
-			}
-			if messageSettings, ok := (*currentQueue.MediaSettings)[mediaSettingsKeyMessage]; ok {
-				d.Set("media_settings_message", flattenMediaSetting(messageSettings))
-			}
+
+		if currentQueue.MediaSettings != nil && currentQueue.MediaSettings.Call != nil {
```
Comment (carnellj-genesys) [inline genesyscloud/resource_genesyscloud_routing_queue.go:599]: This was a bit of screwery.  The JSON body did not change, but the API owners moved the swagger def from being a map to a specific object with values for each object.  So I had to go through and change some of the code for the routing queues to move away from using a map to using individual objects.

File: genesyscloud/resource_genesyscloud_auth_role.go [hunk 3]
```diff
@@ -234,7 +234,7 @@ func readAuthRole(ctx context.Context, d *schema.ResourceData, meta interface{})
 	log.Printf("Reading role %s", d.Id())
 
 	return withRetriesForRead(ctx, d, func() *resource.RetryError {
-		role, resp, getErr := authAPI.GetAuthorizationRole(d.Id(), nil)
+		role, resp, getErr := authAPI.GetAuthorizationRole(d.Id(), false, nil)
```
Comment (carnellj-genesys) [inline genesyscloud/resource_genesyscloud_auth_role.go:237]: This is the new field call that will disable usercount.  This is an extremely expensive operation and we don't use if for CX as Code

File: go.mod [hunk 4]
```diff
@@ -61,6 +61,7 @@ require (
 	github.com/mitchellh/go-wordwrap v1.0.1 // indirect
 	github.com/mitchellh/mapstructure v1.5.0 // indirect
 	github.com/mitchellh/reflectwalk v1.0.2 // indirect
+	github.com/mypurecloud/platform-client-sdk-go/v94 v94.0.1 // indirect
```
Comment (charliecon) [inline go.mod:64]: Need to run `go mod tidy` to get rid of the dependency. GitHub Actions can't seem to handle having two versions of our platform sdk at the same time

**Review by charliecon — APPROVED** [review]

**Pull request merged**